frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -1,18 +1,136 @@
 
 import { AuthElement } from "../AuthElement";
 import { withCatchAndSuspense } from "../RSErrorBoundary";
+import { USLink } from "../USLink";
+import { showError } from "../AlertNotifications";
 import { MemberType } from "../../hooks/UseOktaMemberships";
+import { validateFileType, validateFileSize } from "../../utils/FileUtils";
+
+import ManagePublicKeyChooseSender from "./ManagePublicKeyChooseSender";
+import ManagePublicKeyUpload from "./ManagePublicKeyUpload";
+
+export const CONTENT_TYPE = "application/x-x509-ca-cert";
+export const FORMAT = "PEM";
+
+function ManagePublicKeySwitchDisplay() {
+    const navigate = useNavigate();
+    const [sender, setSender] = useState("");
+    const [fileContent, setFileContent] = useState("");
+    const [file, setFile] = useState<File | null>(null);
+    const [fileSubmitted, setFileSubmitted] = useState(false);
+
+    // TODO: mocked for now - make the call you need when sending the file
+    const { sendFile } = {
+        sendFile: (data: {
+            contentType?: string;
+            fileContent?: string;
+            file?: File | null;
+        }) => {
+            data = { file: null };
+            return data;
+        },
+    };
+
+    const onSenderSelect = (selectedSender: string) => {
+        setSender(selectedSender);
+    };
+
+    const onBack = () => {
+        navigate(-1);
+    };
+
+    const onPublicKeySubmit = async (
+        event: React.FormEvent<HTMLFormElement>
+    ) => {
+        event.preventDefault();
+
+        if (fileContent.length === 0) {
+            showError("No file contents to validate.");
+            return;
+        }
+
+        try {
+            sendFile({
+                contentType: CONTENT_TYPE,
+                fileContent: fileContent,
+                file: file,
+            });
+            setFileSubmitted(true);
+        } catch (e: any) {
+            console.trace(e);
+            showError(`Uploading public key failed. ${e.toString()}`);
+        }
+    };
+
+    const onFileChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        // No file selected
+        if (!event?.target?.files?.length) {
+            setFile(null);
+            return;
+        }
+
+        const file = event.target.files.item(0);
+        if (!file) return; // so typescript doesnt complain
+
+        const content = await file.text();
+
+        const { fileTypeError } = validateFileType(file, FORMAT, CONTENT_TYPE);
+        if (fileTypeError) {
+            showError(fileTypeError);
+        }
+        const { fileSizeError } = validateFileSize(file);
+        if (fileSizeError) {
+            showError(fileSizeError);
+        }
+
+        if (!fileTypeError && !fileSizeError) {
+            setFile(file);
+            setFileContent(content);
+        }
+    };
+
+    return (
+        <>
+            {sender.length === 0 && (
+                <ManagePublicKeyChooseSender onSenderSelect={onSenderSelect} />
```

Comment:
Can we omit this step for now since we're not going to need Sender granularity in the key?  (We've discussed this already, but just wanted to follow up on it!)

Reply:
Sure, should I add a comment and comment it out or remove entirely?

Reply:
I think you can comment it out and add an inline comment for what it's not currently being used yet -- that'd make the re-introduction of this logic much simpler whenever we have that scope granularity :)